feat: data import api endpoint

ReneSchroederLJ opened the pull request:
## Description

- added an endpoint to import data
- import supports .xlsx files
- data is uploaded per standard (demand, production, deliver and stock)
- existing data for the standard is replaced

solves #899

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files
- [x] If helm chart has been changed, the chart version has been bumped to either next major, minor or patch level (compared to released chart).

tom-rm-meyer-ISST commented on backend/src/main/java/org/eclipse/tractusx/puris/backend/file/logic/service/ExcelService.java:
```diff
@@ -0,0 +1,494 @@
+                Date day = row.getCell(7).getDateCellValue();
+
+                ItemUnitEnumeration unitEnum = ItemUnitEnumeration.fromValue(unitOfMeasurement);
+                DemandCategoryEnumeration categoryEnum = DemandCategoryEnumeration.fromValue(demandCategoryCodeStr.toUpperCase());
```
Incorrect values here lead to 500er status code instead of an error message

ReneSchroederLJ commented on backend/src/main/java/org/eclipse/tractusx/puris/backend/file/logic/service/ExcelService.java:
```diff
@@ -0,0 +1,494 @@
+@Slf4j
+public class ExcelService {
+
+    private final List<String> demandColumns = List.of(
```
While `lastUpdatedOnDateTime` is a property of the SAMM models for these standards it is not a property of their local representations. Only item stocks have this property locally.

ReneSchroederLJ commented:
Thank you for the thorough review.

I addressed most of your comments, but I had to omit the changes to validation in order not to make this PR to complicated. 

I created an [issue](https://github.com/eclipse-tractusx/puris/issues/906) for a validation rework to further improve the quality of error reporting.

tom-rm-meyer-ISST reviewed (changes requested):
Hey, really increased quality here! Some findings - you can also handle them with #906. I attached my working import files.
[demand.xlsx](https://github.com/user-attachments/files/21041231/demand.xlsx)
[stock_soft_fail.xlsx](https://github.com/user-attachments/files/21041232/stock_soft_fail.xlsx)
[production_no_fail.xlsx](https://github.com/user-attachments/files/21041233/production_no_fail.xlsx)
[delivery_no_fail.xlsx](https://github.com/user-attachments/files/21041234/delivery_no_fail.xlsx)

tom-rm-meyer-ISST commented on backend/src/main/java/org/eclipse/tractusx/puris/backend/file/logic/service/ExcelService.java:
```diff
@@ -0,0 +1,494 @@
+                        .build();
+
+                demands.add(demand);
+            } catch (Exception e) {
```
Handled in #906
